docs/manual/signature-checker.tex

```diff
@@ -45,7 +45,7 @@ \section{Signature annotations\label{signature-annotations}}
   A \emph{field descriptor} (\href{https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.3.2}{JVMS \S 4.3.2}), such as
   \<Lpackage/Outer\$Inner;>, is used in a \<.class> file's constant pool,
   for example to refer to other types.  It abbreviates primitives and
-  arrays.  It uses internal form (binary names, but with \</> instead of
+        arrays.  It uses internal form (\emph{ClassGetName}s, but with \</> instead of
```

Comment:
Why does this state that internal form is a variant of ClassGetName rather than a variant of binary name?
[JVMS section 4.2.1](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.2.1) defines internal form, and says it is a variant of binary names.
Is JVMS section 4.2.1 incorrect, or am I misinterpreting it?

Reply:
I asked about it on the asm mailing list and this is the reply I received:

>I've always though that the section 4.1 was confusing, i believe it's because historically, prior to 1.5 i think, you were not able to use an array as an internal name, now you can do a ldc on an array or an invokevirtual clone().
The section 5.1 is more precise of what is an internal name of an array
  https://docs.oracle.com/javase/specs/jvms/se10/html/jvms-5.html#jvms-5.1

So it looks like internal form in a variant on class get name only for non array types. Upon further inquiry I was told that the internal form of `Object` class is `java/lang/Object` and that of an array of `Object`s is `[Ljava/lang/Object;` and that the form `java/lang/Object[]` isn't used in a class file.
The links mentioned in [this](https://github.com/typetools/checker-framework/issues/2008) issue seem to verify these findings.